### PR TITLE
Ensure we have a server name to attempt

### LINKED ISSUE #6256: Get-DBAAgReplica doesn't work using just the AG name [Bug]
I get no results if i provide just the AG name.
Is this expected behavior? It says the instance is not required in the documentation. It says the AG isn't required. One or the other must be?

### Environmental information

<!--
Run below command, paste results below:
& {"``````";"#### PowerShell version:`n$($PSVersionTable | Out-String)"; "`n#### dbatools Module version:`n$(gmo dbatools -List | select name, path, version | fl -force | Out-String)";"``````"} | clip
-->
Name                           Value                                                                                                                                                                                                     
----                           -----                                                                                                                                                                                                     
PSVersion                      5.1.17134.858                                                                                                                                                                                             
PSEdition                      Desktop                                                                                                                                                                                                   
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                   
BuildVersion                   10.0.17134.858                                                                                                                                                                                            
CLRVersion                     4.0.30319.42000                                                                                                                                                                                           
WSManStackVersion              3.0                                                                                                                                                                                                       
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                       
SerializationVersion           1.1.0.1  

#### dbatools Module version:
Note - I use
import-module dbatools -RequiredVersion 1.0.47

Name    : dbatools
Path    : C:\.....WindowsPowerShell\Modules\dbatools\1.0.50\dbatools.psd1
Version : 1.0.50

Name    : dbatools
Path    : C:\.....\WindowsPowerShell\Modules\dbatools\1.0.47\dbatools.psd1
Version : 1.0.47

Name    : dbatools
Path    : C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules\dbatools\dbatools.psd1
Version : 0.9.509




#### SQL Server: 
<!-- Paste output of `SELECT @@VERSION` -->
```sql
/* REPLACE WITH output of @@VERSION */
```
Microsoft SQL Server 2016 (SP2) (KB4052908) - 13.0.5026.0 (X64)   Mar 18 2018 09:11:49   Copyright (c) Microsoft Corporation  Enterprise Edition: Core-based Licensing (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) (Hypervisor) 

Microsoft SQL Server 2012 (SP4) (KB4018073) - 11.0.7001.0 (X64)   Aug 15 2017 10:23:29   Copyright (c) Microsoft Corporation  Enterprise Edition (64-bit) on Windows NT 6.3 <X64> (Build 9600: ) (Hypervisor) 


```

### Report

<!--
Things to consider:
- Errors received are not related to permissions? No
- Have you tried the same command using powershell.exe instead of a hosted powershell instance like ISE or VS Code?

--> 


#### Host used

- [ ] powershell.exe
- [ ] ISE
- [ ] VS Code
- [ ] Other (please specify)

If anything other than powershell.exe was used, please confirm that you can duplicate the issue with powershell.exe

- [ ] Still buggy in powershell.exe

#### Errors Received
No errors. Just no result set. Is it a documentation issue? Is the instance name required?
<!-- 
Run this command and paste below:
& {"``````";$error[0] | select *;"``````"} | clip
-->

#### Steps to Reproduce

<!--
If you have confirmed this issue can be reproduced, please provide the exact steps (T-SQL, PowerShell, ext)
-->

#### Expected Behavior

<!--
What did you expect to happen?
-->
I expected the same results I get when using the instance name, based on the instructions
#### Actual Behavior

<!--
What happened?
-->
I got nothing - tried on multiple AGs, OSs, and SQL Versions

### PR DESCRIPTION
**Pull request by cfradenburg**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #6256 )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
If the call doesn't provide either a servername or an AG object (which includes server) alert the user 

### Approach
In a begin block make sure that one of the following is true:
- a SqlInstance is passed in
- InputObject is passed in
- The function was called via the pipeline

### Commands to test
Call each affected function with neither SqlInstance (or primary for Sync-DbaAvailabilityGroup) nor InputObject
Call each affected function with either parameter
Call each affected function with both parameters
Call each affected function via the pipeline

### Learning
The `begin` block executes before the piped parameter is filled.  In addition to check to see if InputObject was passed in by name I had to check $MyInvocation.ExpectingInput to see if the function was called via the pipeline.

### CODE REVIEW

File: functions/Add-DbaAgDatabase.ps1 [hunk 1]
```diff
@@ -116,6 +116,11 @@ function Add-DbaAgDatabase {
         [switch]$UseLastBackup,
         [switch]$EnableException
     )
+    begin {
+        if (!($SqlInstance -or $InputObject -or $MyInvocation.ExpectingInput)) {
```
Comment (wsmelton) [inline functions/Add-DbaAgDatabase.ps1]: We generally use `Test-Bound SqlInstance -Not` (returns true if the parameter was not provided) or `Test-Bound SqlInstance` will return false if not provided.

I have not had chance to update contributing guide with these bits...

As well, is there a reason `$MyInvocation.ExpectingInput` is needed?

Comment (cfradenburg) [reply functions/Add-DbaAgDatabase.ps1]: Thanks.  I'll update the code to use that instead.  That likely won't be until later next week due to the holidays.  As to `$MyInvocation.ExpectingInput`, check the Learning section above.

Comment (wsmelton) [reply functions/Add-DbaAgDatabase.ps1]: Simple fix there is to simply check if these parameters are provided in the process block and not the Begin...that is the pattern used in all our other commands.

**Comment by potatoqualitee** [comment]
hows this looking? Is it ready to commit? all checks have passed 👍

**Comment by potatoqualitee** [comment]
*poke*

**Comment by cfradenburg** [comment]
Thanks for the poke.  I could have sworn that I put a comment in but I must not have clicked the button.  It's ready to go.

**Comment by wsmelton** [comment]
@cfradenburg you can click the "Ready for Review" button to let us know it is ready to go.

**Comment by potatoqualitee** [comment]
heck yessssssssss thank you both! 💯

**Pull request merged**